pages/dashboard.vue

```diff
@@ -33,42 +32,85 @@
   </v-container>
 </template>
 
-<script>
+<script lang="ts">
+import Vue from 'vue'
 import { SessionStorageKeys } from 'sbc-common-components/src/util/constants'
-// local
+import { mapGetters } from 'vuex'
 import UserProduct from '@/components/UserProduct.vue'
 import { ProductCode, ProductStatus } from '@/enums'
-import { getAccountProducts, getKeycloakRoles, getProductInfo } from '@/utils'
+import { fetchAccountProducts, fetchOrganization, getKeycloakRoles, getProductInfo, sleep } from '@/utils'
 
-export default {
+export default Vue.extend ({
   components: {
-    UserProduct
+    UserProduct,
   },
-  // Called every time before loading page components.
-  // TODO: need to start Token Service to refresh KC token periodically?
-  asyncData(context) {
-    if (!sessionStorage.getItem(SessionStorageKeys.KeyCloakToken)) {
-      context.redirect('/signin')
+  asyncData ({ $config, redirect, store }) {
+    // if user is not logged in, redirect to home page
+    const token = sessionStorage.getItem(SessionStorageKeys.KeyCloakToken)
+    if (!token) {
+      return redirect($config.baseURL)
+    }
+
+    // get roles
+    let roles: string[]
+    try {
+      roles = getKeycloakRoles()
+    } catch {
+      roles = []
+    }
+    store.commit('setRoles', roles)
+
+    // check if user is staff
+    let isStaff: boolean
+    try {
+      isStaff = roles?.includes('staff') || false
+    } catch {
+      isStaff = false
+    }
+    store.commit('setStaff', isStaff)
+
+    // if user is staff, redirect to Business Registry staff dashboard
+    if (isStaff) {
+      return redirect($config.authURL + 'staff/dashboard/active')
     }
   },
-  data() {
+  data () {
     return {
-      getProductInfo,
-      roles: getKeycloakRoles(),
-      isStaffSbc: false,
+      getProductInfo, // for use in template
       subscribedProducts: [],
     }
   },
-  async mounted() {
-    if (this.roles.includes('staff')) {
-      const redirectURL = this.$config.authURL + 'staff/dashboard/active'
-      window.location.href = redirectURL
+  computed: {
+    ...mapGetters(['isSbcStaff', 'getRoles', 'getAccountId']),
+  },
+  async mounted () {
+    // get account id from object in session storage
```

Comment:
Since asyncData() above is run before the account sync (which happens in SbcHeader or SbcAuthMenu), getting the account and id must be done here.